### PR TITLE
Category feature

### PR DESCRIPTION
**Pull request by abhisheksahani-nova**
Files to be reviewed :- ( 100 lines of code approx )

1. src/App.js
2. src/components/Filter/Filter.jsx
3. src/components/Navbar/Navbar.jsx
4. src/context/category-context.js
5. src/pages/authentication-pages/signup/Signup.jsx
6. src/pages/landing-page/ProductCategory/ProductCategory.jsx
7. src/pages/product-listing/ProductListing.jsx
8. src/reducer/filterReducer.js

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -78,9 +75,18 @@ function Navbar() {
           <small>Cart</small>
         </div>
 
-        <NavLink to="/login">
-          <button className="btn nav-auth-btn-style">Login</button>
-        </NavLink>
+        {token ? (
+          <button
+            className="btn nav-auth-btn-style"
+            onClick={() => handleLogout()}
```
Comment (MayankKumar10) [inline src/components/Navbar/Navbar.jsx:81]: you can use onclick without arrow function also

File: src/pages/authentication-pages/signup/Signup.jsx [hunk 2]
```diff
@@ -22,10 +22,7 @@ function Signup() {
...
       try {
         const response = await axios.post("/api/auth/signup", userData);
-        localStorage.setItem(
-          "token",
-          JSON.stringify(response.data.encodedToken)
-        );
+        localStorage.setItem("token", response.data.encodedToken);
         navigate("/");
       } catch (error) {
         console.log(error);
```
Comment (MayankKumar10) [inline src/pages/authentication-pages/signup/Signup.jsx:31]: remove console log

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions everything else looks good to me you can merge it

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me. Like that you have reduced redundant code.

File: src/components/Navbar/Navbar.jsx (hunk 1 above)
Comment (abhisheksahani-nova) [reply src/components/Navbar/Navbar.jsx:81]: Yahh I can , its just out of habit

File: src/pages/authentication-pages/signup/Signup.jsx (hunk 2 above)
Comment (abhisheksahani-nova) [reply src/pages/authentication-pages/signup/Signup.jsx:31]: I have kept that to see errors , since I have not added , toast yet

**Pull request merged**